REFERENCE.md

```diff
@@ -0,0 +1,265 @@
+
+> TODO Note Andrew: Suggest we bring up more clearly the distinction between row- and set-level validations, following with how custom_sql_expr/statement (and built-in rules) support them
+
+For both `CUSTOM_SQL_EXPR` and `CUSTOM_SQL_STATEMENT`, the templated parameter `$column` in the `custom_sql_expr` and `custom_sql_statement` blocks will be substituted with the `column_id` specified in each `rule_binding`. `CUSTOM_SQL_STATEMENT` additionally supports custom parameterized variables in the `custom_sql_arguments` block. These parameters can be bound separately in each `rule_binding`, allowing `CUSTOM_SQL_STATEMENT` to be defined once to be reused across multiple `rule_binding`s with different input arguments.
+
+The SQL code block in `CUSTOM_SQL_STATEMENT` rules must include the string `from data` in the `custom_sql_statement` block, i.e. it must validate data returned by a `data` common table expression (CTE). The common table expression `data` contains rows returned once all `row_filters` and incremental validation logic have been applied.
+
+> TODO Note Andrew: the user can choose to read from the underlying entity (for ex. when they want to ignore incremental logic. We need to explain various usage scenarios and best practices here
+
+`CloudDQ` reports validation summary statistics for each `rule_binding` and `rule` by appending to the target BigQuery table specified in the CLI argument `--target_bigquery_summary_table`. Record-level validation statistics are captured the in columns `success_count`, `success_percentage`, `failed_count`, `failed_percentage`, `null_count`, `null_percentage`. The rule type `NOT_NULL` reports the count of `NULL`s present in the input `column-id` in the columns `failed_count`, `failed_percentage`, and always set the columns `null_count` and `null_percentage` to `NULL`. `CUSTOM_SQL_STATEMENT` rules do not report record-level validation statistics and therefore will set the content of the columns `success_count`, `success_percentage`, `failed_count`, `failed_percentage`, `null_count`, `null_percentage` to `NULL`.
+
+Set-level validation results for `CUSTOM_SQL_STATEMENT` rules are captured in the columns `complex_rule_validation_errors_count` and `complex_rule_validation_success_flag`. `complex_rule_validation_errors_count` contains the count of rows returned by the `custom_sql_statement` block. `complex_rule_validation_success_flag` is set to `TRUE` if `complex_rule_validation_errors_count` is equals to 0, `FALSE` if `complex_rule_validation_errors_count` is larger than 0, and `NULL` for all other rule types that are not `CUSTOM_SQL_STATEMENT`.
+
+#### Filters
+**Filters**: Defines how each `Rule Binding` can be filtered. The content of the `filter_sql_expr` field will be inserted into a SQL `WHERE` clause for filtering your data to the rows for validation.
```

Comment:
Replace first sentence with "Defines a filtering condition for data extracted from the respective entity to generate a sub-set of data for validation"